src/util/entropyCreateStateFromJsons.ts

```diff
@@ -2,66 +2,95 @@ import { genotypeColors } from "./globals";
 import { defaultEntropyState } from "../reducers/entropy";
 
 type JsonAnnotations = Record<string, JsonAnnotation>
-// enum Strand {'+', '-'} // other GFF-valid options are '.' and '?'
-type Strand = string;
-type JsonSegmentRange = {start: number, end: number}; // Start is 1-based, End is 1-based closed (GFF)
+
+type Strand = '+' | '-' // other GFF-valid options are '.' and '?'
+
+interface JsonSegmentRange {
+  /** 1-based */
+  start: number
+
+  /** 1-based closed (GFF) */
+  end: number
+}
+
 interface JsonAnnotation {
   /* Other properties are commonly set in the JSON structure, but the following are
   the only ones read by Auspice */
-  end?: number;
```

Comment:
...aren't these trailing semicolons required by Typescript? [The docs](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#interfaces) suggest they are.

Reply:
They're not required in JavaScript due to ASI. Whether to use them (or commas in this case) is mostly a stylistic choice – ESLint rule page [@stylistic/js/semi](https://eslint.style/rules/js/semi) has a good overview of pros and cons.

As mentioned in the commit message 7e6a7300795abf50b520be9defee387e781f7052, I removed them to improve consistency within the file and with other files (e.g. [phyloTree types](https://github.com/nextstrain/auspice/blob/8a6e56fa4b926de7230bf3bd35fa9e590d878fd5/src/components/tree/phyloTree/types.ts)).

Reply:
I'm only a slight lean towards no semicolons. I'll happily add semicolons to all the TypeScript code I've added if that's what the team desires. My stronger opinion is that we should either enforce stylistic consistency via ESLint rules and not nitpick comments on PRs.

Maybe a topic for dev chat?

Reply:
> My stronger opinion is that we should either enforce stylistic consistency via [ESLint rules](https://github.com/nextstrain/auspice/blob/8a6e56fa4b926de7230bf3bd35fa9e590d878fd5/.eslintrc.yaml#L40-L44) and not nitpick comments on PRs,

In my defense, I was not trying to "nitpick"; I was legitimately confused about why there had been semicolons and then they were removed in a revision to the same PR. (I had also, honestly, managed to forget "ASI" was a thing, and I was happier for it.) 

I also don't really see semicolon usage in Javascript as a "style" issue but that's probably down to having dealt with really hard to figure out bugs caused by semicolon insertion behaving in surprising ways.

Reply:
> I also don't really see semicolon usage in Javascript as a "style" issue but that's probably down to having dealt with really hard to figure out bugs caused by semicolon insertion behaving in surprising ways.

Fair, there are other "style" rules that we've adopted to prevent bugs, such as [eqeqeq](https://eslint.org/docs/latest/rules/eqeqeq). I'm down to add `@stylistic/js/semi: "always"` under the "Code quality rules" section.

Reply:
I just [tried](https://github.com/nextstrain/auspice/compare/8a6e56fa4b926de7230bf3bd35fa9e590d878fd5...0536ee3a0cbc3b0438b03eb77ba4e904b5bfa357) `@stylistic/semi: error` and it doesn't enforce semicolons in this case (within an `interface` definition).

I'm going to close this out and keep as-is, but please re-open if you still think semicolons should be used here.